Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -243,72 +237,81 @@ - (BOOL)startScanningWithResultBlock:(void (^)(NSArray *codes))resultBlock error
     AVCaptureSession *session = nil;
     
     // Configure the session
-    if (!self.hasExistingSession) {
+    if (!self.session) {
         self.captureDevice = [self newCaptureDeviceWithCamera:self.camera];
-        session = [self newSessionWithCaptureDevice:self.captureDevice
-                                              error:error];
+        session = [self newSessionWithCaptureDevice:self.captureDevice error:error];
     }
-    
-    if (session) {
-        self.session = session;
-        self.hasExistingSession = YES;
-        
-        // Configure the rect of interest
-        self.captureOutput.rectOfInterest = [self rectOfInterestFromScanRect:self.scanRect];
-        
-        // Configure the preview layer
-        self.capturePreviewLayer.cornerRadius = self.previewView.layer.cornerRadius;
-        [self.previewView.layer insertSublayer:self.capturePreviewLayer atIndex:0]; // Insert below all other views
-        [self refreshVideoOrientation];
-        
-        // Configure 'tap to focus' functionality
-        [self configureTapToFocus];
-        
-        self.resultBlock = resultBlock;
-        
-        // Start the session after all configurations
+
+    if (!session) {
+        // we rely on newSessionWithCaptureDevice:error: to populate the error
+        return NO;
```

Comment:
seems we would also end up here if `self.session != nil` when entering this method. Would we want to assert and early return in that case?